### PR TITLE
neovim: add config to passthru

### PR DESCRIPTION
**Pull request by teto**
- register the config in the store under the name init.vim which is more
appropriate for neovim.
- generated config added to passthru so that one can easily pass the current config to a raw/unwrapped neovim (helps with development).

For instance, home-manager could reference the config in $XDG_CONFIG_HOME/nvim/init.vim
without the need to wrap nvim with a --cmd.

I develop on neovim and I would like to run the raw binary with the nix generated config but because it keeps changing name it's not always easy. This adds an easy way out.

cc @timokau 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/neovim/wrapper.nix [hunk 1]
```diff
@@ -181,7 +181,7 @@ let
     preferLocalBuild = true;
 
     buildInputs = [makeWrapper];
-    passthru = { unwrapped = neovim; };
+    passthru = { unwrapped = neovim; inherit neovimrcFile; };
```
Comment (timokau) [inline pkgs/applications/editors/neovim/wrapper.nix:184]: Maybe just call this `configFile` in both cases? I don't feel very strongly about this though.

**Review by timokau — APPROVED** [review]
ofBorg eval fails, but other than that looks good to me :+1:

File: pkgs/applications/editors/neovim/wrapper.nix (hunk 1 above)
Comment (teto) [reply pkgs/applications/editors/neovim/wrapper.nix]: I renamed it to configFile.

**Comment by teto** [comment]
thnaks for the review. I had forgotten to cherry-pick some changes from my fork of nixpkgs. It should be good now.

**Comment by teto** [comment]
actually hold on I might have a better idea than this PR. will reopen when I sort it out.

**Comment by teto** [comment]
the problem (in my usage etc) is that the wrapper writes the file. I am preparing a PR with a function genNeovimConfig that returns `{ wrapperFlags, rcContent }` so that the caller can choose wether or not to wrap neovim with `-u init.vim` (which breaks neovim behavior and is not necessary in home-manager, which could write the file in ~/.config/nvim/init.vim

**Comment by teto** [comment]
finally got my `genNeovimConfig` working \o/ will post after cleanup (this WE ?). Got it integrated with my home-manager fork and the experience is much more pleasant.

**Pull request merged**